deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/config/ZKMysqlDataSourceFactory.java

```diff
@@ -84,9 +99,49 @@ public BasicDataSource build() throws Exception {
             host = reader.getHost(replicaSet);
             port = reader.getPort(replicaSet);
         }
-        KnoxDBKeyReader.init((role));
-        String userName = KnoxDBKeyReader.getUserName();
-        String password = KnoxDBKeyReader.getPassword();
-        return DatabaseUtil.createMysqlDataSource(host, port, userName, password, pool);
+
+        if (this.useMTLS) {
+            String userName = getUserNameFromSpiffeId();
+            String password = "password";
+            String trustPassword = this.trustPasswd;
+            String clientPassword = this.clientPasswd;
+            Map<String, String> proxyConnectionProps = ImmutableMap.<String, String>builder()
+                // ssl properties
+                .put("sslMode", "VERIFY_CA" )
+                .put("trustCertificateKeyStoreUrl", "file:/var/lib/normandie/fuse/jkstrust/generic" )
+                .put("trustCertificateKeyStoreType", "JKS" )
+                .put("trustCertificateKeyStorePassword",  trustPassword)
+                .put("clientCertificateKeyStoreUrl", "file:/var/lib/normandie/fuse/jks/generic" )
```

Comment:
Hardcoding the certificate urls also locks them in to only supporting the certs we use at Pinterest. We may also want to make the cert urls configurable.

Could make sense to make the key store type configurable while we're at it.

Reply:
> Hardcoding the certificate urls also locks them in to only supporting the certs we use at Pinterest. We may also want to make the cert urls configurable.
> 
> Could make sense to make the key store type configurable while we're at it.

sounds good.